internal/provisioner/controller/gateway.go

```diff
@@ -272,9 +279,12 @@ func (r *gatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 			}
 
 			// Deployment replicas
-			if gatewayClassParams.Spec.Envoy.WorkloadType == contour_api_v1alpha1.WorkloadTypeDeployment &&
-				gatewayClassParams.Spec.Envoy.Replicas > 0 {
-				contourModel.Spec.EnvoyReplicas = gatewayClassParams.Spec.Envoy.Replicas
+			if gatewayClassParams.Spec.Envoy.WorkloadType == contour_api_v1alpha1.WorkloadTypeDeployment {
+				contourModel.Spec.EnvoyReplicas = gatewayClassParams.Spec.Envoy.Replicas // nolint:staticcheck
```

Comment:
I think we're mising a ">0" check on `gatewayClassParams.Spec.Envoy.Replicas` here? Otherwise by default we're provisioning an Envoy deployment with 0 replicas, instead of using the default from the model (2).

Reply:
So:
```go
if gatewayClassParams.Spec.Envoy.Replicas > 0 {
        contourModel.Spec.EnvoyReplicas = gatewayClassParams.Spec.Envoy.Replicas // nolint:staticcheck
}
```